### PR TITLE
many: start of mini unattended installer

### PR DESCRIPTION
**Pull request by mvo5**
In order to test the installation of Ubuntu Core this commit adds a new Minimal Unattended Installer (muinstaller) that can be used to test the installer API. The initial focus is to test the classic+modes install but it will be able to install core later too.

This also includes a refactor for some gadget code to make it more consumable from the muinstaller.

### CODE REVIEW

File: tests/lib/muinstaller/main.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+		}
+		// XXX: reuse
+		// gadget/install/content.go:mountFilesystem() instead
+		// (it will also call udevadm)
+		if err := mkfs.Make(stru.Filesystem, part.KernelDeviceNode, stru.Label, 0, 0); err != nil {
+			return err
+		}
+
+		// mount
+		mountPoint := runMntFor(stru.Label)
```
Comment (pedronis) [inline tests/lib/muinstaller/main.go:178]: confused, we shouldn't use the same place as snapd no?

File: tests/lib/muinstaller/main.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	chgId, err := cli.InstallSystem(details.Label, opts)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Change %s created\n", chgId)
+	return waitChange(chgId)
+}
+
+// XXX: pass in created partitions instead?
+// XXX2: should we mount or will snapd mount?
```
Comment (pedronis) [inline tests/lib/muinstaller/main.go:148]: we discussed this, snapd will mount

File: tests/lib/muinstaller/main.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+		return err
+	}
+	if output, err := exec.Command("cp", "-a", src, dst).CombinedOutput(); err != nil {
+		return osutil.OutputErr(output, err)
+	}
+
+	return nil
+}
+
+// XXX: or will POST {"action":"install","step":"finalize"} do that?
```
Comment (pedronis) [inline tests/lib/muinstaller/main.go:213]: for sure snapd

File: tests/lib/muinstaller/main.go [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+	}
+	if err := createSeedOnTarget(bootDevice, seedLabel); err != nil {
+		return fmt.Errorf("cannot create seed on target: %v", err)
+	}
+	// XXX: or will POST {"action":"install","step":"finalize"} do that?
+	if err := writeModeenvOnTarget(seedLabel); err != nil {
+		return fmt.Errorf("cannot write modeenv on target: %v", err)
+	}
+	//XXX: will the POST below trigger a reboot on the snapd side? if
+	//     not we need to reboot here
```
Comment (pedronis) [inline tests/lib/muinstaller/main.go:252]: no, we should reboot here

**Comment by mvo5** [comment]
Closing this one in favor of https://github.com/alfonsosanchezbeato/snapd/pull/6 for now to avoid having two moving installer targets. I incoperated the feedack in the other PR

**Pull request closed without merging**